ProcessMaker/Models/ProcessRequestToken.php

```diff
@@ -1152,4 +1171,64 @@ public function reassign($toUserId, User $requestingUser)
         $this->user->notify($notification);
         event(new ActivityAssigned($this));
     }
+
+    /**
+     * Determines the destination URL based on the element destination type specified in the definition.
+     *
+     * @return string|null
+     */
+    public function getElementDestinationAttribute(): ?string
+    {
+        $definition = $this->getDefinition();
+        $elementDestinationProp = $definition['elementDestination'] ?? null;
+        $elementDestination = null;
+        $elementDestinationType = null;
+
+        try {
+            $elementDestinationProp = json_decode($elementDestinationProp, true);
+            if (is_array($elementDestinationProp) && array_key_exists('type', $elementDestinationProp)) {
+                $elementDestinationType = $elementDestinationProp['type'];
+            }
+        } catch (Exception $e) {
+            return null;
+        }
+
+        switch ($elementDestinationType) {
+            case 'customDashboard':
+            case 'externalURL':
+                if (array_key_exists('value', $elementDestinationProp)) {
+                    if (is_string($elementDestinationProp['value'])) {
+                        $elementDestination = $elementDestinationProp['value'];
+                    } else {
+                        $elementDestination = $elementDestinationProp['value']['url'] ?? null;
+                    }
+                }
+                break;
+            case 'taskList':
+                $elementDestination = route('tasks.index');
+                break;
+            case 'homepageDashboard':
+                if (hasPackage('package-dynamic-ui')) {
```

Comment:
for all element destination route maybe we can create a factory

Reply:
The method was improved